### ISSUE TITLE
Purge session ID surrogate key on login and when new messages arrive

### ISSUE THREAD

**Issue by gabestein** [pubpub/pubpub#2958]
### What?
We can implement caching for logged-in visits by creating a separate cache hash and surrogate key within those hashes for each session. To avoid issues when users logout as one user and login as another in one browser profile, and to make sure new messages always show to users, we need to do a few things:

- Append a hashed user ID to the no cache cookie (e.g. `pp-no-cache-<id>`) on both login and new user creation. We need to keep the no-cache text itself so we can target it effectively in Varnish; otherwise we'll have to target absence, which is never predictable given cookie behavior.
- Update the Fastly config to add this cookie's contents to both the content hash and the surrogate key header
- Purge this surrogate key in fastly when the user receives new notification
- Remove the code that purges the surrogate key on logout, since it's unique to each user and thus no longer needed!

### Why?
Make the logged in experience much faster. Reduce logged in load on server.

**Comment by gabestein** [comment]
Okay @tefkah I think here's what's left after #2960:

- I don't think we should be using hashed user IDs on profile pages. It exposes the user ID to anyone who can use Fastly.
- If you create a Pub in a community and add a user attribution to it, that user's logged in profile, both on the community and on www, logged in and out, is not purged. Looking behind the scenes, this appears to be because on profile pages we're no longer sending (or are overwriting?) the community domain surrogate keys.

These both, unfortunately, delay launch. If we can get them fixed tomorrow, however, I feel confident moving forward.

**Comment by tefkah** [comment]
Great, I'll get on it!

1. I agree, if we take the effort to hide them in the cookie it doesn't make sense to expose them there. I propose we use the users slug, as that is already in the url anyway.
2. Whoops, that is kind of silly, i'll try to figure out why that happens. I probably just need to add the community hostname manually.

**Comment by tefkah** [comment]
2. That problem does exist, but I think it's actually related to releases.

Here is the problem workflow

1. Add user which has no pubs in this community to a pub
2. This sends a purge request to the user page 
3. Go to `www.pubpub.org/user/<user-slug>`. Cache miss, but no pub, because it isn't released.
4. Release Pub
5. Go back to `www.pubpub.org/user/<user-slug>`. Still no Pub, because this page isn't purged on release.
6. Remove and re-add user
7. Go back to `www.pubpub.org/user/<user-slug>`. Now the pub shows up.

So the solution is to also purge all the user page on release of a Pub, or any edit to a Pub that affects that little card, soo

Pub: update, delete
PubAttribution (any on the pub): create, update, delete
Release: create

I don't think we need to update the user pages of co-authors when an author updates their profile: when an author updates their profile, it will purge any community that they have pubs in. 
The user pages of co-authors are also tagged with the community of the pub they share with the author, so will automatically get purged too.
